OpenSim/Common/FileSystemPath.h

```diff
@@ -0,0 +1,122 @@
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+// INCLUDES
+#include <OpenSim/Common/osimCommonDLL.h>
+#include "OpenSim/Common/IO.h"
+#include "OpenSim/Common/Object.h"
+#include "Simbody.h"
+
+namespace OpenSim {
+
+//==============================================================================
+//                            OPENSIM FILESYSTEMPATH
+//==============================================================================
+/**
+* A class for handling file system paths. The class holds two variables: 1) a
+* string for the path and 2) a bool _isFile to denote of the path is a file
+* or a directory.
+*
+* For convention, the word "Path" can refer to the FileSystemPath for either
+* a file or directory. "FilePath" and "DirPath" are specific to a file or
+* directory, respectively.
+*
+* This class has methods for resolving paths with respect to some other given
+* directory (i.e. not necessarily the current working directory). In order to
+* resolve these cases, we use methods in SimTK::Pathname. Refer to that class's
+* documentation for the rules of how these are evaluated.
+*
+* @author Carmichael Ong
+*/
+class FileSystemPath : public Object {
+    OpenSim_DECLARE_CONCRETE_OBJECT(FileSystemPath, Object);
+
+public:
+    /** Default constructor **/
+    FileSystemPath();
+
+    /** Construct FileSystemPath from a string. **/
+    FileSystemPath(std::string& pathName);
+    FileSystemPath(std::string& pathName, bool isFile);
+
+    /** Use default copy constructor and assignment operator. */
+    FileSystemPath(const FileSystemPath&) = default;
+    FileSystemPath& operator=(const FileSystemPath&) = default;
+
+    /** Destructor. **/
+    ~FileSystemPath() = default;
+
+    // Get a FileSystemPath for the current working directory.
+    FileSystemPath getWorkingDirPath();
+
+    // Get absolute path to a file or directory. getAbsoluteFilePath() and
+    // getAbsoluteDirPath() checks if the FileSystemPath is a file or dir
+    // first. getAbsolutePath() checks if the FileSystemPath is a file or dir
+    // then calls the corresponding function. If the path was not an absolute 
+    // path, then the directory is resolved with respect to the current working 
+    // directory.
+    FileSystemPath getAbsolutePath();
+    FileSystemPath getAbsoluteFilePath();
+    FileSystemPath getAbsoluteDirPath();
```

Comment:
I'm not sure these last two are necessary as part of the public interface; if you need these methods still, they could be private.